### PR TITLE
Subworkflow bam_variants_demix_boot_freyja allow optional bootstrap skipping

### LINKED ISSUE #5554: [FEATURE] allow skipping freyja boot subworkflow
### Is your feature request related to a problem? Please describe

Allow skipping of the bootstrapping for the freyja subworkflow to save computational time. 

See for more info on discussion https://github.com/nf-core/viralrecon/issues/417

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Joon-Klaps**
## PR checklist

Closes #5554  and modules freyja update

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success},
+                { assert snapshot(workflow.out.variants,
+                                workflow.out.versions).match() },
+                // All depend on a database that get's updated regurlarly
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:40]:
```suggestion
                // All depend on a database that gets updated regularly
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:85]:
```suggestion
                // All depend on a database that gets updated regularly
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:119]:
```suggestion
                // All depend on a database that gets updated regularly
```

**Review by SPPearce — APPROVED** [review]
Looks great to me.

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                input[6] = FREYJA_UPDATE.out.lineages_meta
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success},
+                { assert snapshot(workflow.out.variants,
+                                workflow.out.versions).match() },
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:84]:
```suggestion
                                  workflow.out.versions).match() },
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    test("sarscov2 - bam - skip - val - nodb"){
+
+        when {
+            workflow {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:false ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:102]:
```suggestion
                      file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                input[6] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success},
+                { assert snapshot(workflow.out.variants,
+                                workflow.out.versions).match() },
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:118]:
```suggestion
                                  workflow.out.versions).match() },
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                }
+            }
+        }
+
+        when {
+            workflow {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:false ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:68]:
```suggestion
                      file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                input[6] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success},
+                { assert snapshot(workflow.out.variants,
+                                workflow.out.versions).match() },
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:39]:
```suggestion
                                  workflow.out.versions).match() },
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    tag "freyja/boot"
+
+    test("sarscov2 - bam - val - nodb"){
+
+        when {
+            workflow {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:false ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:23]:
```suggestion
                      file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test [hunk 10]
```diff
@@ -0,0 +1,127 @@
+nextflow_workflow {
+
+    name "Test Workflow BAM_VARIANT_DEMIX_BOOT_FREYJA"
+    script "../main.nf"
+    workflow "BAM_VARIANT_DEMIX_BOOT_FREYJA"
+    tag "subworkflows"
+    tag "subworkflows_nfcore"
+    tag "bam_varaint_demix_boot_freyja"
```
Comment (SPPearce) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/tests/main.nf.test:8]:
```suggestion
    tag "bam_variant_demix_boot_freyja"
```

**Comment by SPPearce** [comment]
Weird, I'm sure that the singularity test was working earlier.

**Comment by Joon-Klaps** [comment]
Yeah, because it's bootstrapping a read selection, of a low coverage sample. This results in some areas containing extremely low coverage or even 0 and the tool crashes. I've reduced the number of bootstrap runs to avoid encountering this issue frequently.

**Comment by Joon-Klaps** [comment]
Thanks @SPPearce!

**Pull request merged**